examples/ssx-test-token-gated/README.md

```diff
@@ -0,0 +1,338 @@
+  chains,
+});
+
+const wagmiClient = createClient({
+  autoConnect: true,
+  connectors,
+  provider,
+});
+```
+
+Wrap your `<App />` component with `<WagmiConfig />`, `<RainbowKitProvider/>` and `<SSXProvider/>` components, making it look like the following:
+
+```tsx
+/* src/index.tsx */
+
+root.render(
+  <React.StrictMode>
+    <WagmiConfig client={wagmiClient}>
+      <RainbowKitProvider chains={chains}>
+        <SSXProvider>
+          <App />
+        </SSXProvider>
+      </RainbowKitProvider>
+    </WagmiConfig>
+  </React.StrictMode>
+);
+```
+
+At `src/App.tsx` some changes are required to hook SSX and RainbowKit. Head to the file and make the following changes:
+
+```tsx
+/* src/App.tsx */
+
+import '@rainbow-me/rainbowkit/styles.css';
+import { useConnectModal, useAccountModal } from '@rainbow-me/rainbowkit';
+import { useSSX } from '@spruceid/ssx-react';
+import { SSXClientSession } from '@spruceid/ssx';
+
+function App() {
+  /* SSX hook */
+  const { ssx } = useSSX();
+  /* RainbowKit ConnectModal hook */
+  const { openConnectModal } = useConnectModal();
+  /* RainbowKit Account modal hook */
+  const { openAccountModal } = useAccountModal();
+  /* Some control variables */
+  const [session, setSession] = useState<SSXClientSession>();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (ssx && loading) {
+      /* Sign-in with SSX whenever the button is pressed */
+      ssx
+        .signIn()
+        .then(session => {
+          console.log(session);
+          setSession(session);
+          setLoading(false);
+        })
+        .catch(err => {
+          console.error(err);
+          setSession(undefined);
+          setLoading(false);
+        });
+    }
+  }, [ssx, loading]);
+
+  useEffect(() => {
+    if (!provider) {
+      setSession(undefined);
+    }
+  }, [provider]);
+
+  const handleClick = () => {
+    /* Opens the RainbowKit modal if in the correct state */
+    if (openConnectModal) {
+      openConnectModal();
+    }
+    /* Triggers the Sign-in hook */
+    setLoading(true);
+  };
+
+  return (
+    <div className="App">
+      <div className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <span>SSX</span>
+      </div>
+      <div className="App-title">
+        <h1>SSX Example Dapp</h1>
+        <h2>Connect and sign in with your Ethereum account</h2>
+      </div>
+      <div className="App-content">
+        {!openConnectModal && provider ? (
+          <>
+            <AccountInfo address={session.address} />
+          </>
+        ) : (
+          <button onClick={handleClick}>SIGN-IN WITH ETHEREUM</button>
+        )}
+        {openAccountModal && ownEnsName && provider ? (
+          <button onClick={openAccountModal}>{session?.address}</button>
+        ) : (
+          <></>
+        )}
+      </div>
+    </div>
+  );
+}
+```
+
+You are now able to run `yarn start` and sign-in using RainbowKit.
```

Comment:
Wasn't able to run this as I got an error because - Provider and ownENSName not defined above yet.

Reply:
@w4ll3 - might be good just to add a quick `yarn install` instruction here as well,